bin/kano-settings-onboot

```diff
@@ -164,132 +282,174 @@ def ensure_correct_browser():
 
+        logger.info('is comment kano_screen_used? {}'.format(has_config_comment('kano_screen_used')))
+        logger.info('is comment kano_screen_used({})? {}'.format(model, get_config_comment('kano_screen_used', model)))
 
-# model and model comment check
-model = screen_data['model']
-logger.info('Screen model: {}'.format(model))
-if '--force' not in sys.argv:
-    check_model_present(model)
+        if not force_mode and (not has_config_comment('kano_screen_used') or \
+                                   get_config_comment('kano_screen_used', model)):
 
-# read status
-status = screen_data['status']
-if not status:
-    sys.exit(1)
+            # The screen is either the same or not set at all
+            logger.info('Explicit HDMI configuration detected, exiting.')
+            sys.exit(1)
+        else:
+            logger.info('New screen was detected, reconfiguring...')
 
-# get edid
-edid = screen_data['edid']
-if not edid:
-    sys.exit(1)
+    # override modes, calculate is_monitor
+    override_models()
+    calculate_is_monitor()
 
+    if verbose:
+        # Explain the new screen settings
+        print 'New status:', json.dumps(status, indent=4)
+        print 'New EDID:', json.dumps(edid, indent=4)
 
-# If the hdmi_group and hdmi_mode values are set explicitely in the config
-# file, skip the detection.
-if get_config_value('hdmi_group') != 0 and get_config_value('hdmi_mode') != 0:
-    if not has_config_comment('kano_screen_used') or \
-       get_config_comment('kano_screen_used', model):
-        # The screen is either the same or not set at all
-        logger.info('Explicit HDMI configuration detected, exiting.')
-        sys.exit()
-    else:
-        # New screen detected, will reconfigure
-        logger.info('New screen was detected.')
+    # fix hdmi audio status
+    if not edid['hdmi_audio'] and is_HDMI():
+        logger.info('hdmi audio not supported on this screen, changing to analogue')
+        set_to_HDMI(False)
 
-# override modes, calculate is_monitor
-override_models()
-calculate_is_monitor()
+    # calculate changes needed
+    changes = compare_and_set_mode() or \
+        compare_and_set_full_range() or \
+        compare_and_set_overscan() or \
+        compare_and_set_optimal_resolution()
 
-# output
-logger.debug(status)
-logger.debug(edid)
+    if changes:
 
-# fix hdmi audio status
-if not edid['hdmi_audio'] and is_HDMI():
-    msg = 'hdmi audio not supported on this screen, changing to analogue'
-    logger.info(msg)
-    set_to_HDMI(False)
+        if verbose:
+            print 'changes applied:', changes
 
+        # write comment to config
+        set_config_comment('kano_screen_used', model)
+        end_config_transaction()
 
-# changes
-changes = compare_and_set_mode() or compare_and_set_full_range() or \
-    compare_and_set_overscan()
+        # reboot
+        if dry_run:
+            logger.debug("dry_run rebooting due to config changes")
```

Comment:
Maybe `dry_run` should output the `config.txt` file that it would write since it isn't actually writing it.